src/main/java/server/Routes.java

```diff
@@ -1,8 +1,21 @@
 package server;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public class Routes {
     private Routes() {
     }
 
-    public static String[] ROUTES = {"/simple_get", "/get_with_body"};
+    public static Map<String, Map<String, String>> ROUTES = new HashMap() {{
```

Comment:
Without the "diamond" operator on your `HashMap`s you are not providing enough type information to the compiler. This objection initialization pattern doesn't really support what you are trying to do here, but this could be replaced with a [static initializer](https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html). Example:

```java
    public static final Map<String, Map<String, String>> ROUTES;
    static {
      Map<String, Map<String, String>> routes = new HashMap<>();
      routes.put("/simple_get", Collections.singletonMap("methods", "GET, HEAD"));
      ...

      ROUTES = Collections.unmodifiableMap(routes);
    }
```

Realistically this example would be fine as-is, but I just want to reinforce the idea that you should try to keep/provide as much type info as possible, and there are likely alternatives if the types are unhappy using a specific pattern.

Reply:
Side note, on a larger project I'd likely end up using Guava for this....
```
    public static final Map<String, Map<String, String>> ROUTES = ImmutableMap.of(
        "/simple_get", Collections.singletonMap("methods", "GET, HEAD")
        ...
    );